skopt/space.py

```diff
@@ -255,7 +258,8 @@ def rvs(self, n_samples=None, random_state=None):
 
     @property
     def transformed_size(self):
-        return len(self.categories)
+        size = len(self.categories)
+        return size if size != 2 else 1
```

Comment:
Okay thanks. Please add an inline comment then, this is really not obvious without knowing.